tests/unit/test_config_provider.py

```diff
@@ -667,11 +717,71 @@ def _create_os_environ_patcher(self):
 
     def test_config_store_deepcopy(self):
         config_store = ConfigValueStore()
-        config_store.set_config_provider('foo', ConstantProvider('bar'))
+        config_store.set_config_provider(
+            'constant_value', ConstantProvider('ABC')
+        )
         config_store_copy = copy.deepcopy(config_store)
-        config_store_copy.set_config_provider('fizz', ConstantProvider('buzz'))
-        assert config_store.get_config_variable('fizz') is None
-        assert config_store_copy.get_config_variable('foo') == 'bar'
+        config_store_copy.set_config_provider(
+            'constant_value_copy', ConstantProvider('123')
+        )
+        assert config_store.get_config_variable('constant_value_copy') is None
+        assert config_store_copy.get_config_variable('constant_value') == 'ABC'
+
+    @pytest.fixture
+    def config_value_store(self):
+        return self._create_config_value_store()
+
+    def test_config_store_providers_not_mutated_after_merge(
+        self, smart_defaults_factory, config_value_store
+    ):
+        sts_regional_endpoints_provider = (
+            config_value_store.get_config_provider('sts_regional_endpoints')
+        )
+        s3_section_provider = config_value_store.get_config_provider('s3')
+        retry_mode_provider = config_value_store.get_config_provider(
+            'retry_mode'
+        )
+
+        sts_regional_endpoints_before = (
+            sts_regional_endpoints_provider.provide()
+        )
+        s3_section_before = s3_section_provider.provide()
+        retry_mode_before = retry_mode_provider.provide()
+
+        smart_defaults_factory.merge_smart_defaults(
+            config_value_store, 'standard', 'some-region'
+        )
+
+        sts_regional_endpoints_after = config_value_store.get_config_variable(
+            'sts_regional_endpoints'
+        )
+        s3_section_after = config_value_store.get_config_variable('s3')
+        retry_mode_after = config_value_store.get_config_variable('retry_mode')
+
+        assert (
+            sts_regional_endpoints_provider.provide()
+            == sts_regional_endpoints_before
+        )
+        assert sts_regional_endpoints_before != sts_regional_endpoints_after
+
+        assert s3_section_provider.provide() == s3_section_before
+        assert s3_section_before != s3_section_after
+
+        assert retry_mode_provider.provide() == retry_mode_before
+        assert retry_mode_before != retry_mode_after
+
+    def test_config_store_providers_not_added_after_copy(self):
```

Comment:
Do we need this test case? It seems like it is already covered here: https://github.com/boto/botocore/blob/1e85087e96651b413f624821ac9226faf21f4547/tests/unit/test_config_provider.py#L405

Reply:
Sure, it was testing adding a different one to another variable, but essentially its the same thing. Removing.